### ISSUE TITLE
[BUG] StyleGAN2: latent vector is ignored

### ISSUE THREAD

**Issue by karray** [labmlai/annotated_deep_learning_paper_implementations#107]
The [implementation](https://github.com/labmlai/annotated_deep_learning_paper_implementations/tree/master/labml_nn/gan/stylegan) of StyleGAN2 does not learn a mapping of the latent vector `z`. The vector `z` is completely ignored, and a variety of generated images is provided by `noise`. To demonstrate the issue, I created a google [collab](https://colab.research.google.com/drive/1SvVPRVMvHCpUZKMotz15DoRKYfw4I0PY) with a pre-trained model that I trained for 55400 iterations.

Images genertd with a random `z` and a **fixed `noise`**:
![image](https://user-images.githubusercontent.com/1911342/152770077-f4d96a1c-9e7c-47ec-bb6a-721d4f4ae282.png)

Images generated with a fixed `z` and **random `noise`**
![image](https://user-images.githubusercontent.com/1911342/152770292-ae25a2e3-863e-4a18-a24f-87e19df2aedf.png)

**Comment by vpj** [comment]
Thanks. Will look into it.

**Comment by Siimarras** [comment]
False or Weakening.
Ignore or Endure

**Comment by lukaboljevic** [comment]
Sorry for commenting on a rather old issue, but I've recently implemented StyleGAN2 (being helped by the code from this repository too) and experienced a similar issue - however, in my case, it wasn't due to `z` or `noise`. 

My models were actually experiencing mode collapse, and no matter what the input was, a pretrained model would always generate the same image. I managed to overcome this issue by playing around with the learning rates and number of gradient accumulation steps.
